pkg/descheduler/descheduler.go

```diff
@@ -75,7 +76,21 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
 		<-ctx.Done()
 		close(stopChannel)
 	}()
-	return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+
+	runFn := func() error {
+		return RunDeschedulerStrategies(ctx, rs, deschedulerPolicy, evictionPolicyGroupVersion, stopChannel)
+	}
+
+	// if deployed as a Deployment,
+	if rs.LeaderElection.LeaderElect {
+		if rs.DeschedulingInterval.Seconds() > 0 {
+			leader.NewLeaderElection(runFn, rsclient, rs.LeaderElection, ctx)
+			return nil
+		}
+		klog.V(1).InfoS("Warning: You set LeaderElection to True but server does not use Deployment kind.")
```

Comment:
technically `DeschedulingInterval` can be set on cronjobs too, so it's not purely indicative of the kind. Maybe reword this error to not imply that (for example a user would be confused if they are using Deployment kind but forgot to set DeschedulingInterval)